### PR TITLE
AP-4840 Fix search process model

### PR DESCRIPTION
**Pull request by dreamind**

### CODE REVIEW

**Comment by dreamind** [comment]
Heaps of changes in SimpleSearchController.java is due to reformatting. There are a couple of files that contains mixes of tabs and spaces for indents.

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/SimpleSearchController.java [hunk 1]
```diff
@@ -64,231 +65,230 @@
...
-  public void setVisibility(HtmlBasedComponent comp, boolean visibility) {
-    String style = comp.getStyle();
-    if (style == null) {
-      style = "";
+
+        doSearchBtn.addEventListener(ON_CLICK, new EventListener<Event>() {
+            public void onEvent(Event event) throws Exception {
+                processSearch();
+            }
+        });
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/SimpleSearchController.java]: I think you could rewrite this (and a bunch of the other event handler registrations) more compactly as
```
doSearchBtn.addEventListener(ON_CLICK, event -> processSearch());
```

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/SimpleSearchController.java [hunk 2]
```diff
@@ -64,231 +65,230 @@
...
+        if (previousSearches == null) {
+            return;
+        }
+
+        List<String> list = new ArrayList<>();
+        for (SearchHistoriesType previousSearch : previousSearches) {
+            if (previousSearch.getSearch().startsWith(comboValue)) {
+                list.add(previousSearch.getSearch());
+            }
+        }
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/SimpleSearchController.java]: Could do this with streams:
```
List<String> list = previousSearches.stream()
                                    .map(SearchHistoriesType::getSearch)
                                    .filter(s -> s.startsWith(comboValue))
                                    .collect(Collectors.toList());
```

**Review by raboczi — APPROVED** [review]
I have some superficial syntax suggestions, but otherwise this seems fine to merge.

**Pull request merged**